src/tools/clang.lua

```diff
@@ -289,3 +289,34 @@
 	function clang.gettoolname(cfg, tool)
 		return clang.tools[tool]
 	end
+
+
+
+--
+-- Compile a string of text with the clang compiler.
+--
+-- @param cfg
+--    The project configuration.
+-- @param text
+--    The source text.
+-- @return
+--    on success the filename of the output object file.
+--    nil on failure.
+--
+
+	function clang.try_compile(cfg, text)
+		-- write the text to a temporary file.
```

Comment:
This shares most of the code with GCC version below, might be a good way to share their code in a common function.